### PR TITLE
Rename extentions to extensions

### LINKED ISSUE #3733: DataBusExtentions misspelled
Should be `DataBusExtensions`. This is a breaking change, should we fix for v6?

Since its extmethods we probably only have to change here

http://docs.particular.net/nservicebus/messaging/databus#configuring-azuredatabus

### PR DESCRIPTION
**Pull request by timbussmann**
fixes #3733 and other usages of "extentions"

Added obsoletes for the renamed classes since but not for the extension methods using/returning the renamed classes. If used as extensions, this shouldn't provide an error, otherwise the obsolete warning should raise the issue?

affected:
* docs PR: https://github.com/Particular/docs.particular.net/pull/1561
* NServiceBus.DataBus.AzureBlobStorage: https://github.com/Particular/NServiceBus.DataBus.AzureBlobStorage/pull/4
* NServiceBus.NHibernate: https://github.com/Particular/NServiceBus.NHibernate/pull/194
* NServiceBus.Persistence.AzureStorage: https://github.com/Particular/NServiceBus.Persistence.AzureStorage/pull/80
* NServiceBus.RavenDB: https://github.com/Particular/NServiceBus.RavenDB/pull/244
* NServiceBus.Newtonsoft.Json: https://github.com/Particular/NServiceBus.Newtonsoft.Json/pull/6

@Particular/nservicebus-maintainers ping

### CODE REVIEW

**Comment by andreasohlund** [comment]
LGTM

File: src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt [hunk 1]
```diff
@@ -695,6 +695,8 @@ namespace NServiceBus
             "ntedException. Will be removed in version 7.0.0.", true)]
         public static void LoadMessageHandlers<T>(this NServiceBus.EndpointConfiguration config, NServiceBus.First<T> order) { }
     }
+    [System.ObsoleteAttribute("Use `LoadMessageHandlersExtensions` instead. Will be removed in version 7.0.0.", true)]
+    public class static LoadMessageHandlersExtentions { }
```
Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: Would this even work? For proper obsoletes you probably need to preserve the extension methods and forward them to the new extension methods. //cc @andreasohlund

**Comment by danielmarbach** [comment]
:+1:

**Comment by andreasohlund** [comment]
Build is failing, need to approve the changes

**Comment by timbussmann** [comment]
@danielmarbach @andreasohlund updated obsoletes and approved API

**Comment by andreasohlund** [comment]
Changes looks good, have you smoke tested any samples? (just to make sure that the obsoletes works properly)

**Comment by timbussmann** [comment]
also fixes https://github.com/Particular/NServiceBus/issues/3734

**Comment by timbussmann** [comment]
@andreasohlund did a quick smoke test on raven and json samples. They have missing methods now since the downstreams aren't updated to provide the config extensions for the renamed class.

**Comment by andreasohlund** [comment]
Ok, lets get this in then @danielmarbach ?

**Pull request merged**